ingester2/src/persist/worker.rs

```diff
@@ -305,7 +305,16 @@ where
     // table in order to make the file visible to queriers.
     let parquet_table_data =
         iox_metadata.to_parquet_file(ctx.partition_id(), file_size, &md, |name| {
-            table_schema.columns.get(name).expect("unknown column").id
+            table_schema
+                .columns
+                .get(name)
+                .unwrap_or_else(|| {
+                    panic!(
+                        "unknown column {name} in table ID {table_id}",
+                        table_id = ctx.table_id()
```

Comment:
```suggestion
                        table_id = ctx.table_id().get()
```

IIRC all our newtype wrappers display as `FooId(123)`, so the panic msg. may be a bit doubled (`...in table ID TableId(123)`).